src/TurbulenceClosures/turbulence_closure_implementations/lagrangian_dynamic_smagorinsky.jl

```diff
@@ -225,6 +225,8 @@ end
     u̅w̅ = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, ℑxyzᶜᶜᶜ, ℑxyzᶠᶠᶠ, uᵢuⱼ, ℑxᶜᵃᵃ, ℑzᵃᵃᶜ, u, w)
     v̅w̅ = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, ℑxyzᶜᶜᶜ, ℑxyzᶠᶠᶠ, uᵢuⱼ, ℑyᵃᶜᵃ, ℑzᵃᵃᶜ, v, w)
 
+    S̅ₘ = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, ℑxyzᶜᶜᶜ, ℑxyzᶠᶠᶠ, ϕ⁰⁵, ΣᵢⱼΣᵢⱼᶜᶜᶜ, u, v, w)
```

Comment:
Currently the code goes into `StackOverflowError`, likely due to the way the filter averaging is done as we might have function compositions which are too deep.

Reply:
we can maybe use a box filter here instead of all this combination of interpolations. 
We can add a filter function to the `struct`

Reply:
Hmm. But this should not be an overflow error, right? Are we hitting some heuristic?

I would try combining steps to see what happens. For example you could define

```julia
ϕ⁰⁵_ΣᵢⱼΣᵢⱼᶜᶜᶜ(i, j, k, grid, u, v, w) = ϕ⁰⁵(i, j, k, grid, ΣᵢⱼΣᵢⱼᶜᶜᶜ, u, v, w)
```

You could also simply write the whole thing manually. With composition this deep, it might also make this code easier to understand.

Reply:
> Does it work if we use a simpler filter?

What is complicated about this?